daemon/volumes_unix.go

```diff
@@ -359,6 +359,9 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 		if err := label.Relabel(bind.Source, container.MountLabel, shared); err != nil {
 			return err
 		}
+		if err := volume.FixUIDGID(container.Config.User, bind.Source, bind.Mode); err != nil {
```

Comment:
I insist that it will work only with numerical uid and probably not well too. Because `--user` is user inside container. You're calling your func on host.
Integration tests doesn't cover username/groupname and failure in resolving as well.